src/commonMain/kotlin/com/faithlife/oauth/AuthorizationHeaderValue.kt

```diff
@@ -260,11 +260,12 @@ sealed class AuthorizationHeaderValue {
                         // e.g. %3A
                         append('%')
 
-                        if (byte < 16) {
+                        val unsigned = byte.toUByte()
+                        if (unsigned < 16u) {
                             append('0')
                         }
 
-                        append(byte.toString(16).uppercase())
+                        append(unsigned.toString(16).uppercase())
```

Comment:
Would `append(String.format("%02x", byte))` be simpler and reuse existing functionality to format a hex number?

Reply:
`String.format` doesn't exist in a multi-platform implementation yet (https://youtrack.jetbrains.com/issue/KT-25506). I suspect because it hasn't been implemented in the LLVM backend yet (https://youtrack.jetbrains.com/issue/KT-42973).

I considered deferring to the iOS/Android layer for the platform specific implementations, but this seems simpler to me.